_posts/2025-04-19-navigating-pagination-with-hybrid-query.md

```diff
@@ -0,0 +1,229 @@
+2. The coordinator node receives 120 total results (60 for the match query and 60 for the k-NN query) and processes them using hybrid search techniques, including normalization and score combination.
+3. When applying hybrid search techniques, any duplicate results appearing in both subqueries are merged into a single result.
+4. The final results are trimmed according to top 10 results ( because of `size = 10`), starting from the 5th entry (because of `from = 5`).
+
+This process ensures that hybrid search efficiently merges and ranks results from multiple subqueries while respecting pagination constraints.
+
+## Why pagination_depth matters
+
+Traditional queries (like match and term) use a simple `from` + `size` formula to determine the maximum number of search results to retrieve from each shard. However, applying this same formula to hybrid search creates challenges that affect result accuracy, or _ground truth_. Ground truth refers to information from direct observation rather than inference; in practice, it represents the best available test or benchmark under reasonable conditions. This serves as a foundation for validating data accuracy, whether through customer feedback, human labels, or the most reliable available testing methods even when classification may be imperfect.
+
+Let's examine an example. Consider a hybrid query search request with these parameters:
+- `from = 0` 
+- `size = 3`
+
+The request contains two subqueries under the hybrid clause: a match query and a k-NN query. The normalization processor is configured to assign a greater weight to the match query (0.7) higher than the k-NN query (0.3), as shown in the following diagram.
+
+![higher-weightage](/assets/media/blog-images/2025-04-19-navigating-pagination-with-hybrid-query/higher-weightage.png){:class="img-centered"}
+
+### How the from + size formula affects ground truth
+
+Modifying `from` or ``size`` values increases the number of results retrieved for each subquery from a shard. However, these new results don't simply append to the last page. They can appear earlier in the ranked list, changing the ground truth you're trying to paginate through. This happens in the following two main scenarios:
+
+1. **Higher-weighted subquery effect**:
+    - New results from a higher-weighted subquery may rank higher after score combination.
+    - As a result, these new entries may appear earlier in the final result set---either on the first few pages or somewhere in the middle---rather than at the end, disrupting the expected pagination order.
+
+    In our example, increasing size to 4 causes an unexpected result: the new Query 1 result (document ID 14) appears as the fourth document instead of at the end, because of Query 1's higher weight during normalization, as shown in the following diagram.
```

Comment:
```suggestion
    In our example, increasing `size` to `4` causes an unexpected result: the new Query 1 result (document ID 14) appears as the fourth document instead of at the end because of Query 1's higher weight during normalization, as shown in the following diagram.
```